components/espsense/espsense.h

```diff
@@ -101,8 +103,8 @@ class ESPSense : public Component {
       ESP_LOGE("ESPSense", "Failed to start UDP listener!");
     }
   }
-  
-  void addPlug(ESPSensePlug plug) {
+
+  void addPlug(ESPSensePlug *plug) {
```

Comment:
How about using the hardware MAC by default if it's the first plug and no specific MAC is configured, and otherwise using the name hash? Like the following:
```suggestion
  void addPlug(ESPSensePlug *plug) {
    if (plug->mac.empty()) {
      if (plug_count == 0) { 
        // First plug to be added, and no MAC set, so default to own hardware
        plug->set_mac_address(get_mac_address_pretty());
      } else {
        // Generate a fake MAC address from the name to prevent issues when there are multiple "plugs" with the same MAC address
        uint32_t name_hash = fnv1_hash(plug->name);
        uint8_t *hash_pointer = (uint8_t *)&name_hash;
        char mac[20];
        sprintf(mac, "%02X:%02X:%02X:%02X:%02X:%02X", 53, 75, hash_pointer[0], hash_pointer[1], hash_pointer[3], hash_pointer[4]);
        plug->set_mac_address(mac);
      }
    }
```
*to caveat any errors…I compiled the above but didn't actually try it on hardware yet as it got a little late!

Reply:
Also forgot, this requires including `esphome/core/helpers.h` as well.

Reply:
Updated